library_information_validator.py

```
@@ -0,0 +1,495 @@
+                logger.error('The index sequence \"{}\" in pool {} contains invalid characters.'\
+                ' Allowed characters: A/T/C/G/N/a/t/c/g/n/-'
+                .format(index, pool_name))
+                quit()
+
+            # check that indexes within a pool have minimum diversity
+            for i in range(index_count,len(index_seq)):
+                if lev.distance(index.lower(), index_seq[i].lower()) < LibrarySheet.MAX_DISTANCE:
+                    logger.warning('The index sequences {} and {} in pool {}'\
+                    ' display low diversity (only {} nt difference).'\
+                    .format(index,index_seq[i], pool_name, lev.distance(index.lower(), index_seq[i].lower())))
+                    warning_low_div += 1
+            index_count += 1
+
+        # checks index length
+        warning_index_length = 0
+        warning_index_balance = 0
+        if(not sindex):
+            index_length = []
+            for index in index_seq:
+                index_length.append(len(index))
+            count_length = Counter(index_length)
+
+            if(len(count_length) > 1):
+                logger.warning('There are {} different index lengths in pool {}, please double check the sequences.'\
+                .format(len(count_length),pool_name))
+                warning_index_length += 1
+
+            # checks color balance in the pool
+            max_length = sorted(count_length.keys())[-1]
+            min_length = sorted(count_length.keys())[0]
+            index_list_colour = []
+            for index in index_seq:
+                index_colour = index.replace('T','G').replace('A','R').replace('C', 'R')
+                index_list_colour.append(list(index_colour))
+
+            for row_nr in range(0,max_length):
+                column = []
+                for row in index_list_colour:
+                    try:
+                        column.append(row[row_nr])
+                    except IndexError:
```

Maybe you should motivate why you catch and ignore this error.

---

Yes, and log - even if it's a `debug` level log. `pass` is bad style 😉